Peer.GitHub/GitHubRequestFetcher.cs

```diff
@@ -68,29 +83,64 @@ public async Task<IEnumerable<PullRequest>> GetPullRequestsAsync(CancellationTok
                     prThreadPages.Where(pr => pr.ReviewThreads.PageInfo.HasNextPage).ToList();
             }
 
-            return prs.Values.Select(pr => pr.Into());
+            foreach (var value in prs.Values.Select(x => x.Into()))
+            {
+                yield return value;
+            }
         }
 
-        private async Task<GraphQLHttpRequest> GenerateInvolvesRequest()
+        private async IAsyncEnumerable<PRSearch.PullRequest> QueryGithubPullRequests(QueryType type, [EnumeratorCancellation] CancellationToken token)
         {
-            var searchParams = new PRSearch.SearchParams(
-                await _username,
-                _config.Orgs,
-                _config.ExcludedOrgs,
-                _config.Count);
+            var cursor = null as string;
+
+            while (!token.IsCancellationRequested)
+            {
+                var response = await _gqlClient.SendQueryAsync<GQL.SearchResult<PRSearch.Result>>(await GenerateRequest(type, cursor), token);
 
-            return new GraphQLHttpRequest(PRSearch.Search.GenerateInvolves(searchParams));
+                if (response.Errors != null)
+                {
+                    Console.WriteLine($"ERROR: {string.Join('\n', response.Errors.Select(x => x.Message))}");
```

Comment:
Logger?

Reply:
We don't have one registered atm. Was going to but want to chat with you about what we went with. Ideally something minimal